public/main/client/wallet.js

```diff
@@ -3,54 +3,50 @@
 const { aes256cbcIv, sha256 } = require('./crypto')
 const settings = require('electron-settings')
 
-const getActiveWallet = () => settings.get('user.activeWallet')
+const getActiveWallet = () => settings.getSync('user.activeWallet')
 
 const setActiveWallet = activeWallet =>
-  settings.set('user.activeWallet', activeWallet)
+  settings.setSync('user.activeWallet', activeWallet)
 
-const getWallets = () => Object.keys(settings.get('user.wallets'))
+const getWallets = () => Object.keys(settings.getSync('user.wallets'))
 
 const getWalletId = seed => sha256.hash(seed)
 
 const getWalletAddresses = walletId =>
-  Object.keys(settings.get(`user.wallets.${walletId}.addresses`))
+  Object.keys(settings.getSync(`user.wallets.${walletId}.addresses`))
 
 const findWalletId = address =>
-  Object.keys(settings.get('user.wallets')).find(walletId =>
+  Object.keys(settings.getSync('user.wallets')).find(walletId =>
     getWalletAddresses(walletId).includes(address)
   )
 
 function getSeed (walletId, password) {
   if (!walletId) {
     return new Error('Origin address not found')
   }
-  const encryptedSeed = settings.get(`user.wallets.${walletId}.encryptedSeed`)
+  const encryptedSeed = settings.getSync(`user.wallets.${walletId}.encryptedSeed`)
   return aes256cbcIv.decrypt(password, encryptedSeed)
 }
 
 const getSeedByAddress = (address, password) =>
   getSeed(findWalletId(address), password)
 
 const setAddressForWalletId = (walletId, address) =>
-  Promise.resolve(
-    settings.set(`user.wallets.${walletId}.addresses`, {
-      [address]: {
-        index: 0
-      }
-    })
-  )
+  settings.setSync(`user.wallets.${walletId}.addresses`, {
```

Comment:
Don't we need to return a `Promise` here and down there in `setSeed()`?

Reply:
the only place where it was used no longer requires this method to return a Promise as I changed the code here https://github.com/autonomoussoftware/metronome-wallet-desktop/pull/484/files#diff-41d41eaf929d89252a265cbc34ab37eeb37bb62b7a9c90477aaa8be54f5ee5a7R28

it makes more sense as it is really a sync method 😅